### PR TITLE
Support default values in `Table::get`

### PR DESCRIPTION
**Pull request by reyk**
This implements support for default values in `Table::get`, see #59

The `if o == 0` to detect the default value matches what is done in the `flatbuffers` crate.

Test cases are currently missing but it would be really nice to get a test framework as suggested by @magnet #42

### CODE REVIEW

**Comment by lovesegfault** [comment]
@reyk Please rebase and force push so the CI re-runs; I fixed it in master :)

File: butte-build/src/codegen.rs [hunk 1]
```diff
@@ -461,7 +462,12 @@ impl ToTokens for ir::Table<'_> {
                 &quote!(butte::ForwardsUOffset),
                 true,
             );
-
+            let default_value = if let Some(default_value) = default_value {
+		let default_value = to_default_value(&ty_wrapped, &default_value);
```
Comment (cpcloud) [inline butte-build/src/codegen.rs]: The indentation in this `let` looks off by 2.

Comment (cpcloud) [reply butte-build/src/codegen.rs]: Though this particular block doesn't seem to correctly format with `cargo fmt`.

File: butte/src/table.rs [hunk 2]
```diff
@@ -55,13 +55,17 @@ where
     pub fn get<'a, T: Follow<'a> + 'a>(
         &'a self,
         slot_byte_loc: VOffsetT,
+        default: Option<T::Inner>,
     ) -> Result<Option<T::Inner>, Error> {
         let v_offset = self.vtable()?.get(slot_byte_loc)?;
         if let Some(v_offset) = v_offset {
             let o = v_offset as usize;
+            if o == 0 {
```
Comment (cpcloud) [inline butte/src/table.rs]: `o_O` not really a huge fan of the `o == 0` spelling, can we change `o` to `offset`? :)

**Review by cpcloud — CHANGES_REQUESTED** [review]
Small formatting nit, otherwise LGTM.

File: butte-build/src/codegen.rs (hunk 1 above)
Comment (reyk) [reply butte-build/src/codegen.rs]: My editor inserted two tabs instead of spaces; fixed.

File: butte/src/table.rs (hunk 2 above)
Comment (reyk) [reply butte/src/table.rs]: Sure

**Review by lovesegfault — APPROVED** [review]

**Review by cpcloud — APPROVED** [review]
Merging. Thanks @reyk !

**Pull request merged**